paywall/src/utils/validators.js

```diff
@@ -27,6 +29,94 @@ export const isAccountOrNull = val => {
   return val === null || isAccount(val)
 }
 
+export const isValidIcon = icon => {
+  if (typeof icon !== 'string') {
+    console.error('The paywall config\'s "icon" property is not a string.')
+    return false
+  }
+  if (
+    icon &&
+    !isURL(icon, {
+      allow_underscores: true,
+      allow_protocol_relative_urls: true,
+      disallow_auth: true,
+    }) &&
+    !isDataURI(icon)
+  ) {
+    console.error('The paywall config\'s "icon" property is not a valid URL.')
+    return false
+  }
+  return true
+}
+
+export const isValidCTA = callToAction => {
+  const callsToAction = [
+    'default',
+    'expired',
+    'pending',
+    'confirmed',
+    'noWallet',
+  ]
+
+  const ctaKeys = Object.keys(callToAction)
+
+  if (ctaKeys.length > callsToAction.length) {
+    console.error(
+      'The paywall config\'s "callToAction" properties contain an unexpected entry.'
+    )
+    return false
+  }
+  if (ctaKeys.filter(key => !callsToAction.includes(key)).length) {
+    // TODO: log which key is bad, or remove this check
+    console.error(
+      `The paywall config's "callToAction" properties contain an unexpected entry.`
+    )
+    return false
+  }
+  if (ctaKeys.filter(key => typeof callToAction[key] !== 'string').length) {
+    console.error(
+      `The paywall config's "callToAction" properties contain an entry whose value is not a string.`
+    )
+    return false
+  }
+
+  return true
+}
+
+export const isValidConfigLocks = configLocks => {
+  if (typeof configLocks !== 'object') {
+    console.error(`The paywall configs's "locks" field is not an object.`)
+    return false
+  }
+  const locks = Object.keys(configLocks)
+  if (!locks.length) return false
+  if (
+    locks.filter(lock => {
```

Comment:
I know this is more of a reshift, but this is a bit wild. How would you feel about seeing if we can extract this whole thing into its own method?

Reply:
Reasonable. I'll do that now.